src/parsers/multitest.js

```diff
@@ -1,29 +1,26 @@
-const PROTRACTOR_SHARDED = /------------------------------------/g
-const SPECFILE_REG = /.+Specs:\s(.*\.js)/g
+const PROTRACTOR_MULTITEST = /------------------------------------/g
+const SPECFILE_REG = /Specs:\s(.*\.js)/g
 
 export default {
-  name: 'ShardedParser',
+  name: 'MultiTestParser',
   test (output) {
-    return PROTRACTOR_SHARDED.test(output) && SPECFILE_REG.test(output)
+    return PROTRACTOR_MULTITEST.test(output) && SPECFILE_REG.test(output)
   },
 
   parse (failedSpecs, output) {
     let match = null
-    let testsOutput = output.split('------------------------------------')
-    testsOutput.shift()
-    let RESULT_REG = /\d+\sspec|assertions?,\s(\d+)\sfailures?/g
+    let testsOutput = output.split('------------------------------------').slice(1)
+    let RESULT_REG = /,\s0 failures/g
     testsOutput.forEach(function (test) {
       let specfile
       let result = 'failed'
       // retrieve specfile from run;
       while (match = SPECFILE_REG.exec(test)) { // eslint-disable-line no-cond-assign
         specfile = match[1]
       }
-      // check for string 'X specs, X failures' and verify that failures === 0;
+      // check for string '0 failures' and then marks the test as passed
       while (match = RESULT_REG.exec(test)) { // eslint-disable-line no-cond-assign
```

Comment:
Hmm, I don't know if I understand the distinction. Protractor Flake will re-run all the tests if there's a non successful exit code and there are [no failedSpecs](https://github.com/NickTomlin/protractor-flake/blob/reorganize-parsers/src/index.js#L52).

I prefer B. because it doesn't treat any of the test groupings as special, so just in case a block of `----` does not contain a matching specfile regex, there's no chance of adding an `undefined` to the `failedSpecs` set.

Let me know if i've missed something though 😄